src/utilities/createResolvedRoute.ts

```diff
@@ -1,34 +1,26 @@
 import { markRaw } from 'vue'
-import { Route, Param, Resolved, routeDepthSymbol, isRejectionSymbol } from '@/types'
+import { Route, Param, ResolvedRoute } from '@/types'
 
-type ResolvedRouteProperties<T extends Route> = {
-  matched: T,
+type ResolvedRouteProperties = {
+  matched: Route,
   matches: Route[],
   name: string,
   path: string,
   query: string,
   params: Record<string, Param[]>,
   depth: number,
-  isRejection?: true,
+  isRejection: boolean,
 }
 
-export function createResolvedRoute<T extends Route>(route: ResolvedRouteProperties<T>): Resolved<T> {
+export function createResolvedRoute(route: ResolvedRouteProperties): ResolvedRoute {
```

Comment:
I decided to keep this only because currently we're using it as a place to mark the components as raw.